ros/src/computing/planning/decision/packages/decision_maker/nodes/decision_maker/decision_maker_node_decision.cpp

```diff
@@ -56,43 +68,62 @@ double DecisionMakerNode::calcIntersectWayAngle(const autoware_msgs::Lane &lanei
     const geometry_msgs::Pose InPose = laneinArea.waypoints.front().pose.pose;
     const geometry_msgs::Pose OutPose = laneinArea.waypoints.back().pose.pose;
 
-    diff = calcPosesAngleDiff(InPose, OutPose);
+    diff = amathutils::calcPosesAngleDiffDeg(InPose, OutPose);
   }
 
   return diff;
 }
 
-bool DecisionMakerNode::isLocalizationConvergence(double _x, double _y, double _z, double _roll, double _pitch,
-                                                  double _yaw)
+bool DecisionMakerNode::isLocalizationConvergence(const geometry_msgs::Point& _current_point)
 {
-  static amathutils::point a;
-  static amathutils::point b;
-
   static std::vector<double> distances;
   static uint32_t distances_count = 0;
+  static geometry_msgs::Point prev_point;
+  static const int param_convergence_count_ = 10;
 
   bool ret = false;
 
-  a.x = b.x;
-  a.y = b.y;
-  a.z = b.z;
-
-  b.x = _x;
-  b.y = _y;
-  b.z = _z;
+  // if current point is not set, localization is failure
+  if (_current_point.x == 0 && _current_point.y == 0 && _current_point.z == 0 && prev_point.x == 0 &&
+      prev_point.y == 0 && prev_point.z == 0)
+  {
+    return ret;
+  }
 
-  distances.push_back(amathutils::find_distance(&a, &b));
-  if (++distances_count > param_convergence_count_)
+  distances.push_back(amathutils::find_distance(prev_point, _current_point));
+  if (++distances_count > param_convergence_count_) /* num of count to judge convergence*/
   {
     distances.erase(distances.begin());
     distances_count--;
     double avg_distances = std::accumulate(distances.begin(), distances.end(), 0) / distances.size();
-    if (avg_distances <= param_convergence_threshold_)
+    if (avg_distances <= 2) /*meter*/
     {
-      ret =  ctx->setCurrentState(state_machine::DRIVE_STATE);
+      ret = true;
     }
   }
-  
+
+  prev_point = _current_point;
   return ret;
 }
+bool DecisionMakerNode::isArrivedGoal()
+{
+  const auto goal_point = current_status_.finalwaypoints.waypoints.back().pose.pose.position;
+
+  if (amathutils::find_distance(goal_point, current_status_.pose.position) < goal_threshold_dist_)
+  {
+    if (current_status_.velocity <= goal_threshold_vel_)
+    {
+      return true;
+    }
+  }
+  return false;
+}
+bool DecisionMakerNode::handleStateCmd(const uint64_t _state_num)
```

Comment:
Future tasks should be specified in the function of pull request or managed as isue.

Reply:
@yk-fujii 
Is it ok to remove it?

Reply:
I removed this function because it is not used.